src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java

```diff
@@ -48,19 +48,22 @@ public static void deleteProductSyncTestData(@Nonnull final SphereClient ctpClie
     }
 
     /**
-     * Deletes all products from the CTP project defined by the {@code ctpClient}.
+     * Unpublishes all published products, then deletes all products from the CTP project defined by the
+     * {@code ctpClient}.
      *
-     * @param ctpClient defines the CTP project to delete the categories from.
+     * @param ctpClient defines the CTP project to delete the products from.
      */
     public static void deleteAllProducts(@Nonnull final SphereClient ctpClient) {
-        final Function<Product, SphereRequest<Product>> unpublishAndDelete = product -> {
-            if (product.getMasterData().isPublished()) {
-                product = ctpClient.execute(ProductUpdateCommand.of(product, Unpublish.of()))
-                                   .toCompletableFuture().join();
-            }
-            return ProductDeleteCommand.of(product);
-        };
-        queryAndApply(ctpClient, ProductQuery::of, unpublishAndDelete);
+        unPublishAllPublishedProducts(ctpClient);
+        queryAndApply(ctpClient, ProductQuery::of, ProductDeleteCommand::of);
+    }
+
+    private static void unPublishAllPublishedProducts(@Nonnull final SphereClient ctpClient) {
+        final Function<Product, SphereRequest<Product>> unPublish = product ->
+                ProductUpdateCommand.of(product, Unpublish.of());
+        final ProductQuery publishedProductsQuery = ProductQuery.of()
+                .withPredicates(p -> p.masterData().isPublished().is(true));
+        queryAndApply(ctpClient, () -> publishedProductsQuery, unPublish);
```

Comment:
Not sure this approach improves readability:
  - `unPublish` is declared as lambda, `publishedProductsQuery` is declared as statement and later converted to lambdla
  - does it really has sense to declare local lambda and use it immediately in the next statement? I'd say it would be more readable if you make them as, for example, static methods, and refer in `queryAndApply` call as references to the class methods.